### PR TITLE
Add MPICH 3.4.1

### PR DESCRIPTION
**Pull request by leofang**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@brianv0 I'd like to generate the patch on the fly in `build.sh`. Could you share with us how you generated it for Apple M1?

**Comment by hhslepicka** [comment]
@leofang should we get this merged to proceed with the tests and pursue the `patch on the fly` later or do you prefer to tackle it all on this PR?

**Comment by leofang** [comment]
> @leofang should we get this merged to proceed with the tests and pursue the `patch on the fly` later or do you prefer to tackle it all on this PR?

We can get it merged for now, though I am not (yet) a maintainer so we'd have to wait anyway 😅

@conda-forge-admin, please rerender

**Comment by hhslepicka** [comment]
@leofang that makes sense to wait for merging... I was just asking because of https://github.com/conda-forge/mpich-feedstock/pull/58#issuecomment-770002237 and I was wondering if you were waiting to make the final version with the `patch on the fly`

**Comment by leofang** [comment]
@conda-forge/mpich let's merge this

**Comment by isuruf** [comment]
see https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/ about `gnuconfig` package and how to patch config.sub and config.guess

**Comment by leofang** [comment]
Cool, thanks @isuruf. There are multiple places needing `config.guess` and `config.sub` in the source code, do I copy them to all those places?

**Comment by isuruf** [comment]
Yes

**Comment by leofang** [comment]
Great, it worked @isuruf! Thanks.

**Comment by leofang** [comment]
@conda-forge/mpich This is ready! PTAL.

File: recipe/build-mpi.sh [hunk 1]
```diff
@@ -11,6 +11,16 @@ export CC=$(basename "$CC")
...
+    # use Conda-Forge's Arm64 config.guess and config.sub, see
+    # https://conda-forge.org/blog/posts/2020-10-29-macos-arm64/
+    list_config_to_patch=$(find ./ -name config.guess | sed -E 's/config.guess//')
+    for config_folder in $list_config_to_patch; do
+        echo "copying config to $config_folder ...\n"
+        cp -v $BUILD_PREFIX/share/gnuconfig/config.* $config_folder
+    done
+fi
+
 ./autogen.sh
```
Comment (dalcinl) [inline recipe/build-mpi.sh:24]: @leofang Do we need to run this unconditionally?

Comment (leofang) [reply recipe/build-mpi.sh:24]: Looks like it was added for OS X Arm64. I can move it to the `if` block above it.

Comment (leofang) [reply recipe/build-mpi.sh]: Done in 06d2e09.

**Comment by leofang** [comment]
@dalcinl CI passed.

**Pull request merged**